beacon_node/beacon_chain/src/attestation_rewards.rs

```diff
@@ -52,12 +52,20 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
             .get_state(&state_root, Some(state_slot))?
             .ok_or(BeaconChainError::MissingBeaconState(state_root))?;
 
+        let block_root = state.get_latest_block_root(state_root);
+        let execution_optimistic =
+            self.is_optimistic_or_invalid_block_root(state_slot, &block_root)?;
```

Comment:
@michaelsproul `is_optimistic_or_invalid_block_root` is currently not used anywhere. Maybe a hint to use something else and delete `is_optimistic_or_invalid_block_root`.

`is_optimistic_or_invalid_block_root` may not be appropriate here, since it will return `BeaconChainError::ForkChoiceError` for finalized block roots, which this route is meant to support

Reply:
@jimmygchen @dapplion what changes are required here? Can you explain a bit more about this.

Reply:
@Gua00va can you test this branch requesting a finalized block root? I believe your implementation will break in that case returning an error